core/trino-main/src/main/java/io/trino/server/DynamicFilterService.java

```diff
@@ -158,15 +159,23 @@ public void registerQuery(
             Set<DynamicFilterId> lazyDynamicFilters,
             Set<DynamicFilterId> replicatedDynamicFilters)
     {
-        Map<DynamicFilterId, SettableFuture<Void>> lazyDynamicFilterFutures = lazyDynamicFilters.stream()
-                .collect(toImmutableMap(filter -> filter, filter -> SettableFuture.create()));
         dynamicFilterContexts.putIfAbsent(queryId, new DynamicFilterContext(
                 session,
                 dynamicFilters,
-                lazyDynamicFilterFutures,
+                lazyDynamicFilters,
                 replicatedDynamicFilters));
     }
 
+    public void registerQueryRetry(QueryId queryId)
```

Comment:
Are we sure `registerQuery` will be called before `registerQueryRetry`? `io.trino.execution.SqlQueryExecution#start` is async in nature

Reply:
Query retry registration should happen via `SqlQueryScheduler#start` which gets called after `registerDynamicFilteringQuery` in `SqlQueryExecution#start`. So it looks ok to me but will let @arhimondr confirm.

Reply:
Yeah, it looks like `SqlQueryScheduler#registerQuery` is called even before `SqlQueryScheduler` is created